server/controllers/chartsController/index.js

```diff
@@ -1,80 +1,156 @@
-import { ObjectID } from 'mongodb'
-
+import { ObjectId } from 'mongodb'
+import deepEqual from 'deep-equal'
 import { collections } from '../../utils/mongoCollections'
 import { SITE_NAMES } from '../../utils/siteNames'
 import {
   generateStudyTargetTotals,
   isAnyTargetIncluded,
-  N_A,
   postProcessData,
   processData,
   processTotals,
-  TOTALS_STUDY,
+  mongoQueryFromFilters,
+  sanitizeQueryParams,
 } from './helpers'
+import {
+  N_A,
+  TOTALS_STUDY,
+  STUDIES_TO_OMIT,
+  DEFAULT_CHART_FILTERS,
+  EMPTY_VALUE,
+  INCLUSION_EXCLUSION_CRITERIA_FORM,
+} from '../../constants'
 
-const STUDIES_TO_OMIT = ['files', 'combined']
-
-export const graphDataController = async (dataDb, userAccess, chart_id) => {
+export const graphDataController = async (
+  dataDb,
+  userAccess,
+  chart_id,
+  parsedQueryParams
+) => {
   const labelMap = new Map()
   const dataMap = new Map()
+  const areFiltersRequested =
+    !!Object.keys(parsedQueryParams).length &&
+    !deepEqual(DEFAULT_CHART_FILTERS, sanitizeQueryParams(parsedQueryParams))
+  const filters = areFiltersRequested
+    ? sanitizeQueryParams(parsedQueryParams)
```

Comment:
we're calling `sanitizeQueryParams` twice, could be DRY this up? Also, it'd be great to not have to sanitize the parsed query params. I think it's worth changing the value of the filter query param if it means we don't need this function.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/248/commits/147bf8a9965b532f34bb176e1e651f339c9bf387